includes/course-theme/class-sensei-course-theme-quiz.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * File containing Sensei_Course_Theme_Quiz class.
+ *
+ * @package sensei-lms
+ * @since 3.15.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Renders the Quiz grade results block.
+ */
+class Sensei_Course_Theme_Quiz {
+
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Sensei_Course_Theme_Quiz constructor. Prevents other instances from being created outside of `self::instance()`.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		if ( 'quiz' !== get_post_type() ) {
+			return;
+		}
+
+		$this->maybe_add_quiz_results_notice();
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function maybe_add_quiz_results_notice() {
+
+		$lesson_id   = Sensei_Utils::get_current_lesson();
+		$quiz_id     = \Sensei()->lesson->lesson_quizzes( $lesson_id );
+		$user_id     = get_current_user_id();
+		$quiz_status = Sensei_Utils::user_lesson_status( $lesson_id, $user_id )->comment_approved;
+
+		// If not one of the statuses that we handle then bail.
+		if ( ! in_array( $quiz_status, [ 'ungraded', 'graded', 'passed', 'failed' ], true ) ) {
+			return '';
+		}
+
+		// Prepare title.
+		$grade = Sensei_Quiz::get_user_quiz_grade( $lesson_id, $user_id );
+		$title = sprintf(
+			// translators: The placeholder is the quiz grade.
+			__( 'Your Grade: %1$s%%', 'sensei-lms' ),
+			$grade
+		);
+		if ( 'ungraded' === $quiz_status ) {
+			$title = __( 'Awaiting grade', 'sensei-lms' );
+		}
+
+		// Prepare message.
+		$text = __( "You've passed the quiz and can continue to the next lesson.", 'sensei-lms' );
+		if ( 'ungraded' === $quiz_status ) {
+			$text = __( 'Your answers have been submitted and your teacher will grade this quiz shortly.', 'sensei-lms' );
+		} elseif ( 'failed' === $quiz_status ) {
+			$passmark = absint( get_post_meta( $quiz_id, '_quiz_passmark', true ), 2 );
+			$text     = sprintf(
+				// translators: The first placeholder is the minimum grade required, and the second placeholder is the actual grade.
+				__( 'You require %1$s%% to pass this quiz. Your grade is %2$s%%.', 'sensei-lms' ),
+				$passmark,
+				$grade
+			);
+		}
+
+		$actions = [];
+
+		// Prepare complete lesson button.
+		$should_show_complete_lesson = true;
```

Comment:
Have you found some case where the complete button shows up? If not, I think we could remove it. WDYT?

Reply:
You're probably right. I was simply following the designs. I thought maybe there is some case that I don't know about when the button shows up.

Removed it. a869ed244f91d253f4675137b0e50f49fb3fe1b2